File: arry_ele_freq_31.pl
#!/usr/bin/perl

### Array element frequency ###
print("Enter a string: ");
$str=<STDIN>;
chomp ($str);
@array=split(/\s+/,$str);
%hash;
foreach $word (@array){
	$hash{$word}++;
}
foreach $ele (keys %hash){
	push @num, $hash{$ele};
#	print("$ele:->$hash{$ele}\n");
}
%count;
foreach $data (@num){
	$count{$data}++;
}

foreach $val (keys %count){
	printf("Frequency $val=> Words: ");
	foreach $elem (keys %hash){
	$object=$hash{$elem};
	if($val == $object){
	printf("$elem ");
	}
	}
print("\n");
}

File: PYTHON/arry_ele_freq_31.py
#!/usr/bin/env python3

# Array element frequency
str_input = input("Enter a string: ")
array = str_input.split()

hash_dict = {}
for word in array:
	hash_dict[word] = hash_dict.get(word, 0) + 1

num = []
for word in hash_dict:
	num.append(hash_dict[word])

count = {}
for data in num:
	count[data] = count.get(data, 0) + 1

for val in sorted(count.keys()):
	print(f"Frequency {val}=> Words: ", end="")
	for elem in hash_dict:
		if val == hash_dict[elem]:
			print(elem, end=" ")
	print()
